### ISSUE TITLE
Error 500 instead of 404 if a controller is named testController

### ISSUE THREAD

**Issue by GuillaumeLeclerc** [balderdashy/sails#2873]
Hello,

I just found an interesting bug. When I make an empty controller calling any method returns error 404. But if I rename it to testController then I get an error 500, if the name is not an integer (/test/foo for example) : 

```
Error (E_UNKNOWN) :: Encountered an unexpected error
: ER_BAD_FIELD_ERROR: Unknown column 'NaN' in 'where clause'
```

Best

**Comment by masumeEbhami** [comment]
really this is a bug or not?

**Comment by sailsbot** [comment]
Thanks for posting, @GuillaumeLeclerc.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
